contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1791,3 +1795,204 @@ List
+    list_free(indexoidlist);
+	pfree(table_name_cpy);
+    return ft_index_name;
+}
+
+/*
+ * is_unique_index
+ * Check if given index is unique index of a relation specified by OID
+ */
+bool
+is_unique_index(Oid relid, const char *index_name)
+{
+    Relation        relation = RelationIdGetRelation(relid);
+    List            *indexoidlist = RelationGetIndexList(relation);
+	ListCell		*cell;
+    bool            is_unique = false;
+	int				unique_key_count = 0;
+
+    foreach(cell, indexoidlist)
+    {
+        Oid indexOid = lfirst_oid(cell);
+        char *name = get_rel_name(indexOid);
+
+        if (strcmp(name, index_name) == 0)
+        {
+            HeapTuple       indexTuple = SearchSysCache1(INDEXRELID, ObjectIdGetDatum(indexOid));
+            Form_pg_index   indexForm = (Form_pg_index) GETSTRUCT(indexTuple);
+
+			/* Check if the key column is unique and is of single-key */
+            if (indexForm->indisunique && indexForm->indrelid == relid && indexForm->indnkeyatts == 1)
+            {
+				/* Check if the key column is non-nullable */
+                for (int i = 0; i < indexForm->indnatts; i++)
+                {
+					AttrNumber attnum = indexForm->indkey.values[i];
+                    if (attnum != 0)
+                    {
+                        HeapTuple attTuple = SearchSysCache2(ATTNUM,
+												ObjectIdGetDatum(relid),
+                                                Int16GetDatum(attnum));
+						if(HeapTupleIsValid(attTuple))
+						{
+							Form_pg_attribute attForm = (Form_pg_attribute) GETSTRUCT(attTuple);
+
+							if (attForm->attnotnull)
+							{
+								unique_key_count++;
+								if (unique_key_count > 1)
+									break;
+							}
+						}
+                        ReleaseSysCache(attTuple);
+                    }
+                }
+
+                if (unique_key_count == 1)
+					is_unique = true;
+			}
+
+            ReleaseSysCache(indexTuple);
+            break;
+        }
+    }
+
+    RelationClose(relation);
+    list_free(indexoidlist);
+    return is_unique;
+}
+
+char
+*gen_createfulltextindex_cmds(const char *table_name, const char *schema_name, const List *column_name, const char *index_name)
+{
+	StringInfoData query;
+
+	initStringInfo(&query);
+
+	/*
+	 * We prepare the following query to create a fulltext index.
+	 *
+	 * CREATE INDEX <index_name> ON <table_name> 
+	 * USING gin(to_tsvector('fts_contains_simple', <column_name>));
+	 *
+	 */
+	appendStringInfo(&query, "CREATE INDEX \"%s\" ON ", index_name);
+	if (schema_name == NULL || strlen(schema_name) == 0 || *schema_name == '\0')
+		appendStringInfo(&query, "\"%s\"", table_name);
+	else
+		appendStringInfo(&query, "\"%s\".\"%s\"", schema_name, table_name);
+
+	appendStringInfo(&query, "USING GIN(");
+	for (int i = 0; i < list_length(column_name); i++)
+	{
+		char *col_name = (char *) list_nth(column_name, i);
+		// Add column name
+		appendStringInfo(&query, "to_tsvector('fts_contains_simple', \"%s\")", col_name);
```

Comment:
It's not a good idea to use string to pass a internal real cmd.

Reply:
Agreed, in the next release I plan to rework this to just generate the correct parse tree within gram-tsql-epilogue.y.c